cli/cycpp.py

```diff
@@ -1568,10 +1578,10 @@ def impl(self, ind="  "):
 
             if key in BUFFERS:  # buffer state, skip
                 continue
-            if 'derived_init' in info:  # derived state, skip
+            if 'internal' in info:  # derived state, skip
                 continue
 
-            opt = True if 'default' in info else False
+            opt = True if 'default' in info or 'derived_init' in info else False
```

Comment:
We need dev's to be explicit. The only time optional tags should be provided is if there is a default value (therefore, setting it is optional).